src/test/java/uk/gov/hmcts/ccd/domain/service/createcase/SubmitCaseTransactionCaseAccessGroupTest.java

```diff
@@ -0,0 +1,800 @@
+        doReturn(state).when(caseTypeService).findState(caseTypeDefinition, "SomeState");
+        doNothing().when(caseDocumentService).attachCaseDocuments(anyString(), anyString(), anyString(), anyList());
+
+        CaseDetails caseDetailsWithCaseAccessGroup = submitCaseTransaction.submitCase(event,
+            caseTypeDefinition,
+            idamUser,
+            caseEventDefinition,
+            inputCaseDetails,
+            IGNORE_WARNING,
+            null);
+
+        verify(caseDocumentService).attachCaseDocuments(anyString(), anyString(), anyString(), anyList());
+        assertCaseDataCaseAccessGroup(caseDetailsWithCaseAccessGroup);
+    }
+
+    private Map<String, JsonNode> organisationPolicyCaseData(String role, String organisationId)
+        throws JsonProcessingException {
+
+        JsonNode data = MAPPER.readTree(""
+            + "{"
+            + "  \"Organisation\": {"
+            + "    \"OrganisationID\": " + organisationId + ","
+            + "    \"OrganisationName\": \"OrganisationName1\""
+            + "  },"
+            + "  \"OrgPolicyReference\": null,"
+            + "  \"OrgPolicyCaseAssignedRole\": \"" + role + "\""
+            + "}");
+
+        Map<String, JsonNode> result = new HashMap<>();
+        result.put("OrganisationPolicyField", data);
+        return result;
+    }
+
+    private Map<String, JsonNode> caseAccessGroupCaseData(String caseAccessGroupType, String caseAccessGroupID)
+        throws JsonProcessingException {
+
+        CaseAccessGroup caseAccessGroup = CaseAccessGroup.builder().caseAccessGroupId(caseAccessGroupID)
+            .caseAccessGroupType(caseAccessGroupType).build();
+
+        CaseAccessGroupWithId caseAccessGroupForUI = CaseAccessGroupWithId.builder()
+            .caseAccessGroup(caseAccessGroup)
+            .id(UUID.randomUUID().toString()).build();
+
+        List<CaseAccessGroupWithId> caseAccessGroupForUIs = new ArrayList<>();
+        caseAccessGroupForUIs.add(caseAccessGroupForUI);
+
+        CaseAccessGroup caseAccessGroup1 = CaseAccessGroup.builder()
+            .caseAccessGroupId("SomeJurisdictionCIVIL:bulk: [RESPONDENT02SOLICITOR]:$ORG$")
+            .caseAccessGroupType("CCD:all-cases-access").build();
+
+        CaseAccessGroupWithId caseAccessGroupForUI1 = CaseAccessGroupWithId.builder()
+            .caseAccessGroup(caseAccessGroup)
+            .id(UUID.randomUUID().toString()).build();
+
+        caseAccessGroupForUIs.add(caseAccessGroupForUI1);
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode data  = mapper.convertValue(caseAccessGroupForUIs, JsonNode.class);
+        Map<String, JsonNode> result = new HashMap<>();
+        result.put(CaseAccessGroups, data);
+        return result;
+    }
+
+    private void assertCaseDataCaseAccessGroup(final CaseDetails caseDetails) {
+        assertAll("Assert Casedetails, Data, CaseAccessGroups",
+            //() -> assertThat(caseDetails.getData().get(CaseAccessGroups), isNotNull()),
```

Comment:
if we do not need this assert, can we remove it and then we do not need the assertAll either.